frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/movement-animator.ts

```diff
@@ -4,33 +4,31 @@ import type OlMap from 'ol/Map';
 import type RenderEvent from 'ol/render/Event';
 import type { Feature } from 'ol';
 import { getVectorContext } from 'ol/render';
-import type Point from 'ol/geom/Point';
 import type { UUID } from 'digital-fuesim-manv-shared';
-import { isArray, isEqual } from 'lodash-es';
-import type { LineString } from 'ol/geom';
-import type { Coordinate } from 'ol/coordinate';
-
-export type Coordinates<T extends LineString | Point> = T extends Point
-    ? Coordinate
-    : Coordinate[];
-
-interface CoordinatePair<T extends Coordinate | Coordinate[]> {
-    startPosition: T;
-    endPosition: T;
-}
+import { isEqual } from 'lodash-es';
+import type {
+    Coordinates,
+    CoordinatePair,
+    GeometryWithCoordinates,
+} from './geometry-helper';
 
 /**
  * Animates the movement of a feature to a new position.
  */
-export class MovementAnimator<T extends LineString | Point> {
+export class MovementAnimator<T extends GeometryWithCoordinates> {
     /**
      * The time in milliseconds how long the moving animation should take
      */
     private readonly movementAnimationTime = 200;
 
     constructor(
         private readonly olMap: OlMap,
-        private readonly layer: VectorLayer<VectorSource>
+        private readonly layer: VectorLayer<VectorSource>,
+        private readonly getNextPosition: (
```

Comment:
```suggestion
        private readonly interpolatePositions: (
```

`getNextPosition` is undescriptive (also in `GeometryHelper`)

Reply:
I've renamed them to `interpolateCoordinates`.